lib/samson/build_finder.rb

```diff
@@ -45,9 +45,10 @@ def find_or_create_builds
       all = []
 
       wait_for_build_creation do |last_try|
+        possible = possible_builds
```

Comment:
some general efficiency, no need to fetch these multiple times when we look for multiple builds